StorageUtility/Windows/StorageUtility.cpp

```diff
@@ -18,9 +18,47 @@ limitations under the License.
 #include <memory>
 #include "StorageUtility.h"
 
+#include "Device.h"
+#include "DeviceDataConverter.h"
+
 namespace vtStor
 {
 
+eErrorCode GetDevicePath(String& Path, std::shared_ptr<cDeviceInterface> Device)
+{
+    eErrorCode error = eErrorCode::None;
+
+    std::map<vtStor::eDeviceDataType, void*> deviceData = Device->Data();
+    if (deviceData.end() == deviceData.find(eDeviceDataType::PSP_INTERFACE_DEVICE_DETAIL_DATA))
+    {
+        Path = nullptr;
+        error = eErrorCode::NullPointer;
+    }
+    else
+    {
+        void* rawValue = deviceData[eDeviceDataType::PSP_INTERFACE_DEVICE_DETAIL_DATA];
+        PSP_INTERFACE_DEVICE_DETAIL_DATA devDetailData = cDeviceDataConverter<PSP_INTERFACE_DEVICE_DETAIL_DATA>::ToDeviceData(rawValue);
+        Path = devDetailData->DevicePath;
+    }
+    return(error);
+}
+
+eErrorCode GetStorageDevices(std::vector<std::shared_ptr<cDeviceInterface>>& Devices, eOnErrorBehavior OnErrorBehavior)
```

Comment:
This code is Windows specific, and should be encapsulated in the Windows cDeviceHandle class